source/Weather.cpp

```diff
@@ -70,29 +71,50 @@ const Point &Weather::Origin() const
 
 
 // Create any environmental effects and decrease the lifetime of this weather.
-void Weather::Step(vector<Visual> &visuals)
+void Weather::Step(vector<Visual> &visuals, const Point &center)
 {
 	// Environmental effects are created by choosing a random angle and distance from
 	// their origin, then creating the effect there.
 	double minRange = hazard->MinRange();
 	double maxRange = hazard->MaxRange();
+	double effectMultiplier = currentStrength;
 
-	// Estimate the number of visuals to be generated this frame.
-	// MAYBE: create only a subset of possible effects per frame.
-	int totalAmount = 0;
-	for(auto &&effect : hazard->EnvironmentalEffects())
-		totalAmount += effect.second;
-	totalAmount *= currentStrength;
-	visuals.reserve(visuals.size() + totalAmount);
-
-	for(auto &&effect : hazard->EnvironmentalEffects())
-		for(int i = 0; i < effect.second * currentStrength; ++i)
-		{
-			Point angle = Angle::Random().Unit();
-			double magnitude = (maxRange - minRange) * sqrt(Random::Real());
-			Point pos = origin + (minRange + magnitude) * angle;
-			visuals.emplace_back(*effect.first, std::move(pos), Point(), Angle::Random());
-		}
+	// If a hazard is system-wide, the max range becomes the edge of the screen,
+	// and the number of effects drawn is scaled accordingly.
+	if(hazard->SystemWide())
+	{
+		// Find the larger of the two screen dimensions, and use that as our new
+		// max range.
+		double newMax = 2. * max(Screen::Width(), Screen::Height());
```

Comment:
No. If you create a circle with a radius equal to the largest dimension, since the screen is a rectangle, the edge of the circle will only just touch the middle of the edges of the screen along that largest dimension.